File: frontend/src/lib/types/onchain.ts
export interface ExchangeFlowResponse {
  asset: string;
  latest_netflow: number;
  avg_netflow_7d: number;
  signal: string;
  strength: number;
  trend: string;
}

export interface MVRVResponse {
  asset: string;
  current_mvrv: number;
  signal: string;
  strength: number;
  zone: string;
}

export interface NUPLResponse {
  asset: string;
  current_nupl: number;
  signal: string;
  strength: number;
  phase: string;
}

export interface BTCNetworkStats {
  hash_rate: number | null;
  difficulty: number | null;
  active_addresses: number | null;
  transaction_count: number | null;
  mempool_size: number | null;
}

File: backend/app/schemas/onchain.py
from __future__ import annotations

from typing import Dict, List, Optional

from pydantic import BaseModel


class OnchainEntry(BaseModel):
    timestamp: str
    value: float


class ExchangeFlowResponse(BaseModel):
    asset: str
    latest_netflow: float
    avg_netflow_7d: float
    signal: str
    strength: float
    trend: str


class MVRVResponse(BaseModel):
    asset: str
    current_mvrv: float
    signal: str
    strength: float
    zone: str


class NUPLResponse(BaseModel):
    asset: str
    current_nupl: float
    signal: str
    strength: float
    phase: str


class BTCNetworkStats(BaseModel):
    hash_rate: Optional[float] = None
    difficulty: Optional[float] = None
    active_addresses: Optional[float] = None
    transaction_count: Optional[float] = None
    mempool_size: Optional[float] = None
